### PR TITLE
[WIP] Update to relay 1.4.1

### PR DESCRIPTION
**Pull request by alloy**
This uses my [updated Relay custom Node ID patches](https://github.com/facebook/relay/compare/master...alloy:custom-node-id-redone), which brings in the latest Relay. While it seems to work well, I’m not sure if we should merge it in this repo at this time just before v4. Let’s wait for now, but feel free to give it a spin!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +63,10 @@
   },
   "homepage": "https://github.com/artsy/emission#readme",
   "files": ["index.js", "data", "lib"],
+  "resolutions": {
+    "graphql": "^0.12.3",
```
Comment (alloy) [inline package.json]: Some of our dev deps pull in an older version, which messes with the tree we want, so just overriding it for now.

File: patches/graphql+0.12.3.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+-  name !== '__configs__') {
+-    return new _GraphQLError.GraphQLError('Name "' + name + '" must not begin with "__", which is reserved by ' + 'GraphQL introspection.', node);
+-  }
++  // if (name.length > 1 && name[0] === '_' && name[1] === '_' &&
++  // // Note: this special case is not part of the spec and exists only to
++  // // support legacy server configurations. Do not rely on this special case
++  // // as it may be removed at any time.
++  // name !== '__configs__') {
++  //   return new _GraphQLError.GraphQLError('Name "' + name + '" must not begin with "__", which is reserved by ' + 'GraphQL introspection.', node);
++  // }
```
Comment (alloy) [inline patches/graphql+0.12.3.patch]: graphql-js made this a hard error now 😢 Just disabling it for now.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx`
- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx`
- `src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistNotForSaleArtworksGrid-tests.tsx`
- `src/lib/Components/Home/ArtistRails/__tests__/ArtistRail-tests.tsx`
- `src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/Relay/__tests__/ArtworksRenderer-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx`
- `src/lib/Scenes/Home/Components/ForYou/Components/__tests__/ArtworkCarousel-tests.tsx`
- `src/lib/Scenes/Home/Components/Sales/Relay/__tests__/SalesRenderer-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
We should try make a concerted effort to deal with all these:

```
src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx
src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx
src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistNotForSaleArtworksGrid-tests.tsx
src/lib/Components/Home/ArtistRails/__tests__/ArtistRail-tests.tsx
src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx
src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx
src/lib/Scenes/Favorites/Components/Artworks/Relay/__tests__/ArtworksRenderer-tests.tsx
src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx
src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx
src/lib/Scenes/Home/Components/ForYou/Components/__tests__/ArtworkCarousel-tests.tsx
src/lib/Scenes/Home/Components/Sales/Relay/__tests__/SalesRenderer-tests.tsx
```

**Comment by orta** [comment]
We should try make a concerted effort to deal with all these:

```
src/lib/Components/Artist/Artworks/__tests__/index-tests.tsx
src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistForSaleArtworksGrid-tests.tsx
src/lib/Components/ArtworkGrids/RelayConnections/__tests__/ArtistNotForSaleArtworksGrid-tests.tsx
src/lib/Components/Home/ArtistRails/__tests__/ArtistRail-tests.tsx
src/lib/Components/Inbox/Conversations/__tests__/index-tests.tsx
src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx
src/lib/Scenes/Favorites/Components/Artworks/Relay/__tests__/ArtworksRenderer-tests.tsx
src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx
src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx
src/lib/Scenes/Home/Components/ForYou/Components/__tests__/ArtworkCarousel-tests.tsx
src/lib/Scenes/Home/Components/Sales/Relay/__tests__/SalesRenderer-tests.tsx
```

**Comment by alloy** [comment]
Closing in favour of #982

**Pull request closed without merging**